DecoratorGenerator/Main.cs

```diff
@@ -9,20 +10,14 @@ public class Main : ISourceGenerator
     {
         public void Execute(GeneratorExecutionContext context)
         {
-            var types = context.Compilation.Assembly.GlobalNamespace
-                .GetNamespaceMembers()
-                .SelectMany(n => n.GetTypeMembers())
-                .Where(t => t.GetAttributes()
-                             .Where(@as => @as.AttributeClass.Name == "DecorateAttribute")
-                             .Any());
-
-            var thirdPartyTypes = context.Compilation.Assembly.GetTypeByMetadataName("WrapperList")?.GetMembers()
+            var types = GetAllDecoratedTypes(context.Compilation.Assembly.GlobalNamespace);
+
+            var thirdPartyTypes = GetAllTypes(context.Compilation.Assembly.GlobalNamespace, x => x.Name == "WrapperList").SelectMany(x => x.GetMembers()
```

Comment:
How about keeping support for `WrapperList` file without namespace?

```suggestion
            var wrapperSymbolWithoutNamespace = context.Compilation.Assembly.GetTypeByMetadataName("WrapperList");

            var wrapperListTypes =
                (wrapperSymbolWithoutNamespace == null)
                ? GetAllTypes(context.Compilation.Assembly.GlobalNamespace, x => x.Name == "WrapperList")
                : new[] { wrapperSymbolWithoutNamespace };

            var thirdPartyTypes =
                wrapperListTypes.SelectMany(x => x.GetMembers()
```

Reply:
I'll merge this PR to get the fix in so everyone can benefit. Thanks for your contribution!